### DISCUSSION TITLE
Runtime error

### DISCUSSION THREAD

**Discussion by substrateai** [neon-sunset/Pinecone.NET#27] [Q&A] [answered]
Hi I am testing the pincone client in 7.0 core console app.  When I start the app I get this exception before the code for the pinecone instance is hit.  It looks to be trying to initialize pinecone before it has the key and environment passed in.  Error is below.

System.TypeLoadException: 'Could not load type 'Pinecone.PineconeClient' from assembly 'pinecone, Version=1.0.2.0, Culture=neutral, PublicKeyToken=null'.'

**Comment by neon-sunset** [comment]
This error occurs before the client constructor is called since it is, as question title says, a runtime error. How does the .csproj of your application look like?

Just in case, I double-checked with sample .NET 7 application and could not reproduce the issue:
`PineconeTest.csproj`
```xml
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFramework>net7.0</TargetFramework>
    <ImplicitUsings>enable</ImplicitUsings>
    <Nullable>enable</Nullable>
  </PropertyGroup>

  <ItemGroup>
    <PackageReference Include="Pinecone.NET" Version="1.2.0" />
  </ItemGroup>

</Project>
```
`Program.cs`
```c#
using Pinecone;

const string env = "us-east4-gcp";
const string key = "your-key";

using var pinecone = new PineconeClient(key, env);

foreach (var indexName in await pinecone.ListIndexes())
{
    using var index = await pinecone.GetIndex(indexName);
    Console.WriteLine($"{indexName}: {index}");
}
```

**Comment by neon-sunset** [comment, marked as answer]
It appears you have a wrong package reference. This library does not have a package versioned `1.0.2.0`. Please make sure that your .csproj has one like in my previous comment.

**Comment by substrateai** [comment]
Hi! Thanks for the super fast response to my question.  It turned out to be a package that was to 'pinecone_sdk'.  This apparently created the exception from the above.  Weird.  I removed that package and all is good!  Thanks!
